aip/general/0236.md

```diff
@@ -0,0 +1,321 @@
+- The system must change `state` to and from `ACTIVE` and `INACTIVE` when the
+  experiment is started or stopped, which can only be done by the "preview" and
+  "stop" custom methods.
+- The first time the "preview" custom method is used, the system **must** create
+  `preview_metadata` and do the following: 
+  - It **must** set the `state` to `ACTIVE`
+  - It **must** set a `log_prefix` string, which is a predefined constant
+    hard coded by the system developers. The same value is used for all preview
+    logs for the given resource type. An example is "FirewallPolicyPreviewLog".
+  - It **must** populate the `start_time` with the current time. 
+- When the "stop" custom method is used, the system **must** do the following:
+  - It **must** set the `state` to `INACTIVE`
+  - It **must** populate the `stop_time` with the current time. 
+- `start_time` **must** be updated every time the status is changed to `ACTIVE`.
+
+### APIs
+
+#### create
+
+- Creating a new experiment for preview **must** support the following use
+cases:
+  - Preview a new policy.
+  - Preview an update to an already live policy.
+  - Preview a deletion of a current policy.
+- For the update and delete use cases, the `policy` field in the `experiment`
+  **must** have the full payload of the live policy copied into it, including
+  the name. 
+  - The user **must** update the fields under test to the new intended state to 
+    preview an update.
+  - The user **must** set the `rules` to empty to preview a delete.
+- Previewing a new policy **must** be supported. 
+  - If the system does not support a nested collection without a live policy, 
+    the user **must** create a top-level resource which has the empty rules in
+    order to represent a no-op.
+  - Experiments are added as children of the no-op policy and their effect is
+    previewed.
+- If the system supports multiple previews for this experiment type, calling
+  "create" more than once **must** create multiple experiment resources. These
+  are represented as new subresources until the experimental policy limit is
+  reached.
+
+#### update
+
+- The name inside `policy` **must not** change.
+- The system **may** set the `state` to INACTIVE if the `state` was ACTIVE at
+  the time of an update. This is so the user can easily distinguish between 
+  previews of different fields-under-test.
+
+#### list
+
+- List **must** support filtering on the state of each experiment.
+  - Filtering on `PreviewMetadata` indicates which experiments are actively 
+    previewed.
+
+### Live policy API changes
+
+The following guidance **must** be followed for live resources which use
+experiments as a nested collection for preview.
+
+#### delete
+
+- A delete of the live policy **must** delete all experiments.
+- To maintain the experiments while negating the effect of the live policy, the
+  live policy **must** be changed to a no-op policy instead of using delete.
+
+### Custom method APIs
+
+#### Preview
```

Comment:
```suggestion
#### preview
```

(For consistency with earlier references and the methods above - please adjust all methods below similarly.)